Sources/CircularProgress/CustomButton.swift

```diff
@@ -71,14 +79,19 @@ extension CALayer {
 			return
 		}
 
+		let animationDelegate = AnimationDelegate()
+		animationDelegate.didStopHandler = { [weak self] finished in
+			self?.setValue(color, forKey: keyPath)
+		}
+
 		let animation = CABasicAnimation(keyPath: keyPath)
 		animation.fromValue = value(forKey: keyPath)
 		animation.toValue = color
 		animation.duration = duration
 		animation.fillMode = .forwards
+		animation.delegate = animationDelegate
 		animation.isRemovedOnCompletion = false
 		add(animation, forKey: keyPath)
```

Comment:
You could add something like this extension to simplify it further:

```swift
extension CALayer {
	func add(_ animation: CAAnimation, forKey key: String?, didStopHandler: ((CAAnimation, Bool) -> Void)?) {
		let animationDelegate = AnimationDelegate()
		animationDelegate.didStopHandler = didStopHandler
		animation.delegate = animationDelegate
		add(animation, forKey: key)
	}
}
```

Reply:
Great idea. 👍